alias/src/main/java/software/amazon/kms/alias/CreateHandler.java

```diff
@@ -30,11 +36,45 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
         final ResourceModel model = request.getDesiredResourceState();
 
         return ProgressEvent.progress(model, callbackContext)
+            .then(progress -> {
+                if (progress.getCallbackContext().isPreCreateCheckDone()) {
+                    return progress;
+                } else {
+                    return proxy.initiate("kms::describe-key", proxyClient, model, progress.getCallbackContext())
+                            .translateToServiceRequest(resourceModel ->
+                                    DescribeKeyRequest.builder().keyId(resourceModel.getAliasName()).build())
```

Comment:
To make this call we need to add the permission to this handler at https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-kms/blob/master/alias/aws-kms-alias.json#L37
This can be verified by running the contract tests. I also remember there was one API (CCAPI) which would let us know what permissions are missing on a handler.

Reply:
Additional describeKey permissions are not necessary. If a caller lacks the permissions, then AccessDenied is sufficient to assert that the alias exists.

Reply:
But what is the point of adding a describeKey call if it returns AccessDenied every time? How are we asserting that the alias exists or does not exist? Is an actual check being made here?

Reply:
Regardless of access, describeKey will still return NotFoundException to same-account callers if an alias/key does not exist. Therefore, if we receive AccessDenied, we know that the key exists.